### PR TITLE
use containerized-data-importer-api

### PR DESCRIPTION
**Pull request by mhenriks**
Signed-off-by: Michael Henriksen <mhenriks@redhat.com>

**What this PR does / why we need it**:

the CDI api moved, so let's use the new location

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -118,7 +121,8 @@ replace (
 
 	kubevirt.io/client-go => ./staging/src/kubevirt.io/client-go
 
-	kubevirt.io/containerized-data-importer => kubevirt.io/containerized-data-importer v1.36.0
+	kubevirt.io/containerized-data-importer => kubevirt.io/containerized-data-importer v1.41.0
```
Comment (rmohr) [inline go.mod:124]: Do we also need real code from CDI? Or is this just an artifact from our go.mod file in staging/client-go/go.mod?

Comment (mhenriks) [reply go.mod:124]: Yeah, unfortunately there is other code imported from CDI.  A function for authorizing clone DataVolumes.

https://github.com/kubevirt/containerized-data-importer/tree/main/pkg/clone

Comment (rmohr) [reply go.mod:124]: Ok, thanks!.

**Comment by rmohr** [comment]
I see

```
            "message": "0/2 nodes are available: 1 node(s) didn't match pod anti-affinity rules, 1 node(s) had taint {node.kubernetes.io/disk-pressure: }, that the pod didn't tolerate.",
```

in this job: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/6742/pull-kubevirt-e2e-k8s-1.22-sig-storage/1457963599946321920.

Probably something for flake hunting :)

**Pull request merged**